src/main/java/core/mate/academy/model/Bulldozer.java

```diff
@@ -1,15 +1,48 @@
 package core.mate.academy.model;
 
+import core.mate.academy.service.MachineProducer;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Add some custom fields that could be only in Bulldozer
  * Do not remove no-args constructor
  */
-public class Bulldozer extends Machine {
+public class Bulldozer<T extends Machine> extends Machine implements MachineProducer<T> {
```

Comment:
```suggestion
public class Bulldozer {
```
we shouldn't parametrize pojo classes

Reply:
Ok, I won't parametrize Bulldozer, Truck & Excavator classes. But should they extend Machine class? I believe they should since all of them are machines